File: repaso_parcial/parcial.py
def nombreDeUsuario(ids_usuarios: list[(chr, chr)], id: list[chr]) -> list[chr]:
    res: list = []
    for i in range(len(id)):
        for j in range(len(ids_usuarios)):
            if id[i] == ids_usuarios[j][0]:
                res.append(ids_usuarios[j][1])
    return res


res = nombreDeUsuario([('1','v'),('2','g')], ['2','1'])
print(res)

File: repaso_parcial/parcial.hs
nombreDeUsuario :: [(Char, Char)] -> [Char] -> [Char]
nombreDeUsuario _ [] = []
nombreDeUsuario ids_usuarios (x:xs)= iteraIds ids_usuarios x : nombreDeUsuario ids_usuarios xs

iteraIds :: [(Char, Char)] -> Char -> Char
iteraIds ((id, us): xs) id2 | id == id2 = us
                            | otherwise = iteraIds xs id2